### PR TITLE
{tools}[foss/2024a] s3fs v2024.9.0

### PR DESCRIPTION
**Pull request by lcniel**
(created using `eb --new-pr`)

Requires:
- [x] #21687
- [x] #21688 
- [x] #21695

### CODE REVIEW

File: easybuild/easyconfigs/s/s3fs/s3fs-2024.9.0-foss-2024a.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }),
+    ('botocore', '1.35.36', {
+        'checksums': ['354ec1b766f0029b5d6ff0c45d1a0f9e5007b7d2f3ec89bcdd755b208c5bc797'],
+    }),
+    ('aioitertools', '0.11.0', {
+        'checksums': ['42c68b8dd3a69c2bf7f2233bf7df4bb58b557bca5252ac02ed5187bbc67d6831'],
+    }),
+    ('aiobotocore', '2.15.2', {
+        'checksums': ['9ac1cfcaccccc80602968174aa032bf978abe36bd4e55e6781d6500909af1375'],
+    }),
```
Comment (Micket) [inline easybuild/easyconfigs/s/s3fs/s3fs-2024.9.0-foss-2024a.eb:34]: maybe we move aiobotocore to boto3, drop it from here as well as dropping jmespath, botocore  and just depend on boto3 instead?

also aioitertools 0.12.0

**Comment by lcniel** [comment]
Test report by @lcniel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
vera01-4 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/lcniel/18d2731c7241613e51902998d6405166 for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera11-2 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/lcniel/c5fef7cf95555e8eaeb847c75211ceeb for a full test report.

File: easybuild/easyconfigs/s/s3fs/s3fs-2024.9.0-foss-2024a.eb (hunk 1 above)
Comment (lcniel) [reply easybuild/easyconfigs/s/s3fs/s3fs-2024.9.0-foss-2024a.eb]: I removed unneeded deps, this should be good to go

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen3

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @lcniel!

**Pull request merged**